pkg/apiserver/auth.go

```diff
@@ -36,32 +44,67 @@ func (s *Server) authMiddleware(next http.Handler) http.Handler {
 
 		token := parts[1]
 
-		// TODO: Implement actual JWT validation
-		// This should use jwt-go or similar library to validate the token
-		// Verify signature, expiration, claims, etc.
-		if !s.validateToken(token) {
+		// Validate token using Kubernetes TokenReview API
+		authenticated, serviceAccount, err := s.validateServiceAccountToken(r.Context(), token)
+		if err != nil {
+			log.Printf("Token validation error: %v", err)
+			respondError(w, http.StatusUnauthorized, "UNAUTHORIZED", "Token validation failed")
+			return
+		}
+
+		if !authenticated {
 			respondError(w, http.StatusUnauthorized, "UNAUTHORIZED", "Invalid or expired token")
 			return
 		}
 
-		// Token validation passed, continue processing request
-		next.ServeHTTP(w, r)
+		log.Printf("Authenticated request from service account: %s", serviceAccount)
+
+		// Extract namespace from service account username
+		// Format: system:serviceaccount:<namespace>:<serviceaccount-name>
+		saParts := strings.Split(serviceAccount, ":")
+		var namespace, serviceAccountName string
+		if len(saParts) == 4 && saParts[0] == "system" && saParts[1] == "serviceaccount" {
```

Comment:
could it happen len(saParts) != 4 and saParts[0] != "system", then how should we handle